_posts/2024-12-18-Management-Console-on-OpenShift.adoc

```diff
@@ -0,0 +1,77 @@
+
+You can build such an image on your own based on the official WildFly images available at https://quay.io/repository/wildfly/wildfly[quay.io/wildfly/wildfly] (see "Extending the image"). Another way is to use the pre-built images from https://quay.io/repository/halconsole/wildfly[quay.io/halconsole/wildfly]. These images are mainly meant for HAL development and testing, but already meet these requirements, which makes them suitable for our use case. In particular the images add a management user `admin:admin` and have a list of preconfigured allowed origins.
+
+To add the allowed origin for the public route, we make use of the https://jmesnil.net/weblog/2024/12/11/a-kubectl-plugin-for-wildfly-jboss-cli/[`jboss-cli`] `kubectl` plugin. This plugin makes it straightforward to connect to a WildFly instance running on OpenShift and execute CLI commands. Please visit https://jmesnil.net/weblog/2024/12/11/a-kubectl-plugin-for-wildfly-jboss-cli/ to find out how to install and use the plugin.
+
+== Instructions
+
+The steps below assume you have access to an OpenShift cluster and installed `kubectl` and the `jboss-cli` plugin.
+
+. Create application
++
+[source,shell]
+----
+oc new-app quay.io/halconsole/wildfly
+oc create route edge --service wildfly --port 9990
+----
+
+. Add allowed origin
++
+Use `oc get pods` to find the name of the pod to connect to and `oc get routes` to get the hostname of the public route to the management interface.
++
+[source,shell]
+----
+kubectl jboss-cli -p <pod>
+----
++
+Login using `admin:admin` and execute these CLI commands:
++
+[source,shell]
+----
+/core-service=management/management-interface=http-interface:list-add(name=allowed-origins,value=https://<hostname>)
+reload
+exit
+----
+
+. Open the management console at `+https://<hostname>+` and login using `admin:admin`.
+
+'''
+
+NOTE: As an alternative to adding the allowed origin, you can also use of the latest online version of the management console available at https://hal.github.io/console.
+
+Background: The management console is a single-page application (https://en.wikipedia.org/wiki/Single-page_application[SPA]) without any server side dependencies. As such it can run on its own and connect to an arbitrary management interface. The online version of the console makes use of this fact. See https://hal.github.io/documentation/get-started/#standalone-mode for more details.
+
+. Create the application as above and find the hostname of the public route using `oc get routes`.
+. Open https://hal.github.io/console
+. Add a management interface to the public route:
++
+Give an arbitrary name, select *https* as scheme, enter the hostname of the public route _without_ https and port *80*:
```

Comment:
When I do this and click on `Ping`, nothing happens. Is it normal?

Reply:
Normally that should show the browser login modal and prompt for the username and password. Somehow this does not work for OpenShift. But you are able to login anyway, aren't you?